src/Components/WebAssembly/WebAssembly/src/Infrastructure/JSInteropMethods.cs

```diff
@@ -35,7 +38,15 @@ public static Task DispatchEvent(WebEventDescriptor eventDescriptor, string even
         {
             var renderer = RendererRegistry.Find(eventDescriptor.BrowserRendererId);
             var jsonSerializerOptions = DefaultWebAssemblyJSRuntime.Instance.ReadJsonSerializerOptions();
-            var webEvent = WebEventData.Parse(renderer, jsonSerializerOptions, eventDescriptor, eventArgsJson);
+
+            // JsonSerializerOptions are tightly bound to the JsonContext. Cache it on first use using a copy
+            // of the serializer settings.
+            if (_jsonContext is null)
+            {
+                _jsonContext = new(new JsonSerializerOptions(jsonSerializerOptions));
```

Comment:
Can we move the `ReadJsonSerializerOptions` call inside this `if` block so it's only done once? Super minor detail I know, but helps to clarify that we only respect its value the first time, and it can't change after.